packages/@react-aria/slider/src/useSliderThumb.ts

```diff
@@ -0,0 +1,124 @@
+import {BaseSliderProps, SliderThumbProps} from '@react-types/slider';
+import {chain, focusWithoutScrolling, mergeProps, useDrag1D} from '@react-aria/utils';
+import {computeOffsetToValue} from './utils';
+import {DEFAULT_MAX_VALUE, DEFAULT_MIN_VALUE, DEFAULT_STEP_VALUE, MultiSliderState} from '@react-stately/slider';
+import {HTMLAttributes, useCallback, useEffect} from 'react';
+import {useFocusable} from '@react-aria/focus';
+import {useLabel} from '@react-aria/label';
+
+interface SliderThumbAria {
+  /** Props for the range input. */
+  inputProps: HTMLAttributes<HTMLElement>;
+
+  /** Props for the root thumb element; handles the dragging motion. */
+  thumbProps: HTMLAttributes<HTMLElement>;
+  
+  /** Props for the label element for this thumb. */
+  labelProps: HTMLAttributes<HTMLElement>;
+}
+
+/**
+ * Provides behavior and accessibility for a thumb of a slider component.
+ * 
+ * @param sliderProps Props used for the Slider component.
+ * @param thumbProps Props used for this specific thumb.
+ * @param state Slider state, created via `useMultiSliderState`.
+ * @param trackRef Ref for the track element.
+ * @param inputRef Ref for the range input element.
+ */
+export function useSliderThumb(
+  sliderProps: BaseSliderProps,
+  thumbProps: SliderThumbProps,
+  state: MultiSliderState,
+  trackRef: React.RefObject<HTMLElement>,
+  inputRef: React.RefObject<HTMLInputElement>
+): SliderThumbAria {
```

Comment:
This is starting to be a lot of arguments. What do you think about making an options object instead, and include the necessary props from slider props and thumb props, along with the two refs. The state could continue being a separate argument potentially.

```tsx
interface SliderThumbOptions extends SliderThumbProps {
  maxValue?: number,
  minValue?: number,
  step?: number,
  isReadOnly?: boolean,
  isDisabled?: boolean,
  trackRef: React.RefObject<HTMLElement>,
  inputRef: React.RefObject<HTMLInputElement>
}

useSliderThumb(options: SliderThumbOptions, state: SliderState)
```

Reply:
Ah, agreed making `SliderThumbOptions` is nicer.  However, I'm leaning towards keeping thumb and slider props as separate fields, rather than having it extend `SliderThumbProps`, to be extra explicit:

```typescript
interface SliderThumbOptions {
  sliderProps: BaseSliderProps;
  thumbProps: SliderThumbProps;
  trackRef: React.RefObject<HTMLElement>;
  inputRef: React.RefObject<HTMLInputElement>;
}
```

Reply:
I think we'd prefer to combine the two props objects. I had a suggestion to avoid needing `minValue` and `maxValue` below, and the others could potentially be different per thumb rather than just a global setting. e.g. `isDisabled` and `isReadOnly` are available on individual thumbs, and perhaps `step` may as well eventually.

Reply:
`isReadOnly` and `isDisabled` are actually already part of `SliderThumbProps`...  If we combine, then either we'd need to name the slider-level readonly/disabled differently:

```tsx
interface SliderThumbOptions extends SliderThumbProps {
  isSliderReadOnly?: boolean;
  isSliderDisabled?: boolean;
  step?: number;
  // etc
}
```

Or whoever is calling `useSliderThumb()` would need to take care to merge them:

```tsx
const {...} = useSliderThumb({
    ...thumbProps,
   isReadOnly: thumbProps.isReadOnly || sliderProps.isReadOnly,
   isDisabled: thumbProps.isDisabled || sliderProps.isDisabled,
   step: sliderProps.step,
});
```

This is actually why I originally leaned towards keeping `sliderProps` and `thumbProps` as separate fields in `SliderThumbOptions`.  But happy to go with your preference for either of the above here!

Reply:
Also, I believe you'd still need minValue / maxValue, so that `useSliderThumb` can map from offset position to actual value (which it currently does by calling `computeOffsetToValue` with the sliderProps, though it really just needs min/max/step).

Another possibility is to get the sliderProps from `state` instead of having it passed in explicitly in `SliderThumbOptions`, since `SliderState` already knows about the slider props...

Reply:
> Or whoever is calling useSliderThumb() would need to take care to merge them:

Yep, I think this is what I'd expect. It should be up to whatever is rendering the thumb to decide which thumbs are disabled IMO. We do this in e.g. Radio already, to combine the group prop with the individual prop.

> Another possibility is to get the sliderProps from state

Yeah, we do this in a few places already. Perhaps min/max/step could be stored as attributes on the state so they can be accessed by the thumbs? Or perhaps the state should have a method for converting from a percentage offset to a value between min and max. Maybe `state.setThumbPercent(index, percent)` - would correspond nicely with `getThumbPercent`.

Reply:
Great idea!